core/trino-spi/src/main/java/io/trino/spi/ptf/ScalarArgument.java

```diff
@@ -55,4 +55,34 @@ public Object getValue()
     {
         return value;
     }
+
+    public static Builder scalarArgumentBuilder()
+    {
+        return new Builder();
+    }
+
+    public static final class Builder
+    {
+        private Type type;
+        private Object value;
+
+        private Builder() {}
+
+        public Builder type(Type type)
+        {
+            this.type = type;
+            return this;
+        }
+
+        public Builder value(Object value)
+        {
+            this.value = value;
+            return this;
+        }
+
+        public ScalarArgument build()
+        {
+            return new ScalarArgument(type, value);
```

Comment:
That looks like a typed value. I wouldn't foresee additional parameters here, and so the builder looks like an overkill

Reply:
Not really. At some point, we will allow arbitrary expressions as scalar arguments.